src/families/tezos/bridge/js.ts

```diff
@@ -0,0 +1,268 @@
+// @flow
+import { BigNumber } from "bignumber.js";
+import invariant from "invariant";
+import { TezosToolkit, DEFAULT_FEE } from "@taquito/taquito";
+import {
+  AmountRequired,
+  NotEnoughBalance,
+  NotEnoughBalanceToDelegate,
+  FeeTooHigh,
+  NotSupportedLegacyAddress,
+  InvalidAddressBecauseDestinationIsAlsoSource,
+  RecommendUndelegation,
+} from "@ledgerhq/errors";
+import { validateRecipient } from "../../../bridge/shared";
+import type { CurrencyBridge, AccountBridge } from "../../../types";
+import {
+  makeSync,
+  makeScanAccounts,
+  makeAccountBridgeReceive,
+} from "../../../bridge/jsHelpers";
+import { getMainAccount } from "../../../account";
+import type { Transaction } from "../types";
+import { getAccountShape } from "../synchronisation";
+import { fetchAllBakers, hydrateBakers, isAccountDelegating } from "../bakers";
+import { getEnv } from "../../../env";
+import { signOperation } from "../signOperation";
+import { patchOperationWithHash } from "../../../operation";
+import { log } from "@ledgerhq/logs";
+
+const receive = makeAccountBridgeReceive();
+
+const createTransaction: () => Transaction = () => ({
+  family: "tezos",
+  mode: "send",
+  amount: new BigNumber(0),
+  fees: null,
+  gasLimit: null,
+  storageLimit: null,
+  recipient: "",
+  networkInfo: null,
+  useAllAmount: false,
+  taquitoError: null,
+});
+
+const updateTransaction = (t, patch) => ({ ...t, ...patch });
+
+const getTransactionStatus = async (account, t) => {
+  const errors: {
+    recipient?: Error,
+    amount?: Error,
+    fees?: Error,
+  } = {};
+
+  const warnings: {
+    amount?: Error,
+    feeTooHigh?: Error,
+    recipient?: Error,
+  } = {};
+
+  let estimatedFees = new BigNumber(0);
+
+  invariant(account.tezosResources, "tezosResources is missing");
+  const { tezosResources } = account;
+
+  if (!t.taquitoError) {
+    if (t.mode !== "undelegate") {
+      if (account.freshAddress === t.recipient) {
+        errors.recipient = new InvalidAddressBecauseDestinationIsAlsoSource();
+      } else {
+        const { recipientError, recipientWarning } = await validateRecipient(
+          account.currency,
+          t.recipient
+        );
+
+        if (recipientError) {
+          errors.recipient = recipientError;
+        }
+
+        if (recipientWarning) {
+          warnings.recipient = recipientWarning;
+        }
+      }
+    }
+
+    if (t.recipient.startsWith("KT") && !errors.recipient) {
+      errors.recipient = new NotSupportedLegacyAddress();
+    }
+
+    // no fee / not enough balance already handled by taquitoError
+    // t.fees will always be set
+
+    estimatedFees = t.fees;
+    if (!tezosResources.revealed) {
+      // FIXME
```

Comment:
what is to be fixed?

Reply:
because the estimated fee somehow do not include the reveal fee and the current version did not include the commit in copied below which set the DEFAULT_FEE.REVEAL correctly